tribe-ext-formatted-event-date-shortcode/index.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+* Plugin Name: The Events Calendar Extension: Formatted Event Date Shortcode
+* Description: Output event Start Time or End Time in <a href="http://php.net/manual/en/function.date.php" target="_blank">whatever valid date/time format you want</a>. Example: [tribe_formatted_event_date id=1234 format="F j, Y, \\a\\t g:ia T" timezone="America/New_York"] would output Event ID 1234's start date (which is 11am in America/Chicago) like this: "August 2, 2017, at 12:00pm EST"
+* Version: 1.0.0
+* Extension Class: Tribe__Extension__Formatted_Event_Date_Shortcode
+* Author: Modern Tribe, Inc.
+* Author URI: http://m.tri.be/1971
+* License: GPLv2 or later
+* License URI: https://www.gnu.org/licenses/gpl-2.0.html
+*/
+
+// Created 2016-09-15 for https://theeventscalendar.com/support/forums/topic/_eventstartdate-format/
+// Turned into an Extension by Cliff on 2016-12-07
+
+// Do not load directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+// Do not load unless Tribe Common is fully loaded.
+if ( ! class_exists( 'Tribe__Extension' ) ) {
+   return;
+}
+
+/**
+* Extension main class, class begins loading on init() function.
+*/
+class Tribe__Extension__Formatted_Event_Date_Shortcode extends Tribe__Extension {
+	
+	// Define the shortcode name
+	public $shortcode_name = 'tribe_formatted_event_date';
+	
+	// Array of the possible values from the "Timezone" section of the table at http://php.net/manual/en/function.date.php
+	private $timezone_formats = array(
+		'e',
+		'I',
+		'O',
+		'P',
+		'T',
+		'Z',
+	);
+	
+	// Same as $this->timezone_formats except in regex format, e.g. for preg_match and preg_replace
+	private $timezone_formats_regex = array(
+		'/e/',
+		'/I/',
+		'/O/',
+		'/P/',
+		'/T/',
+		'/Z/',
+	);
+	
+	// Temporary placeholder to be replaced with manually-created timezone format string
+	// Make sure no character is one from http://php.net/manual/en/function.date.php
+	private $timezone_temp_placeholder = '----';
```

Comment:
Would this make more sense as variable inside of `shortcode_output()`? It looks to me like it's only used inside that function.

Either way it's up to you. This is some awesome work and looks ready to merge. If you make this one last change and then merge it I'm sure that's cool as well.